File: __opt/ALGORITHM_Java/Wave File Handler/asciiToWave.java
public class GlobalMembersAsciiToWave
{

	// local headers

	// C++ Language Library headers

	static int Main(int argc, RefObject<String[]> argv)
	{
	  // the splash screen 
	  System.out.print("\n  ***************************************");
	  System.out.print("\n");
	  System.out.print("  ***   ASCII to WAVE FILE CONVERTER  ***");
	  System.out.print("\n");
	  System.out.print("  *** (c) Christoph Lauer Engineering ***");
	  System.out.print("\n");
	  System.out.print("  ***************************************");
	  System.out.print("\n");
	  System.out.print("\n");

	  // test and extract the arguments
	  if (argc != 4)
	  {
		System.out.print("USAGE: asciiToWave [ascii path] [wave path] [sample rate]");
		exit(0);
	  }
	  System.out.print("ASCCI FILE PATH = ");
	  System.out.print(argv.argvalue[1]);
	  System.out.print("\n");
	  System.out.print("WAVE  FILE PATH = ");
	  System.out.print(argv.argvalue[2]);
	  System.out.print("\n");
	  // get the sample rate
	  int sampleRate = Integer.parseInt(argv.argvalue[3]);
	  System.out.print("SAMPLERATE      = ");
	  System.out.print(sampleRate);
	  System.out.print("\n");

	  // call the convert function
	  clauer.io.WaveFileHandler.asciiToWave(argv.argvalue[1], argv.argvalue[2], sampleRate);

	  // give no error back
	  return(0);
	}
}

//----------------------------------------------------------------------------------------
//	Copyright © 2006 - 2009 Tangible Software Solutions Inc.
//
//	This class is used to simulate the ability to pass arguments by reference in Java.
//----------------------------------------------------------------------------------------
final class RefObject<T>
{
	T argvalue;
	RefObject(T refarg)
	{
		argvalue = refarg;
	}
}

File: Wave File Handler/asciiToWave.cpp

// local headers
#include "wave_file_handler.h"

// C++ Language Library headers
#include <iostream>

int main (int argc, char *argv[])
{
  // the splash screen 
  std::cout << "\n  ***************************************" << std::endl;
  std::cout << "  ***   ASCII to WAVE FILE CONVERTER  ***" << std::endl;
  std::cout << "  *** (c) Christoph Lauer Engineering ***" << std::endl;
  std::cout << "  ***************************************" << std::endl << std::endl;
  
  // test and extract the arguments
  if (argc != 4)
  {
    std::cout << "USAGE: asciiToWave [ascii path] [wave path] [sample rate]";
    exit(0);
  }   
  std::cout << "ASCCI FILE PATH = " << argv[1] << std::endl;
  std::cout << "WAVE  FILE PATH = " << argv[2] << std::endl;
  // get the sample rate
  int sampleRate = atoi(argv[3]);
  std::cout << "SAMPLERATE      = " << sampleRate << std::endl;
  
  // call the convert function
  clauer::io::WaveFileHandler::asciiToWave(argv[1], argv[2], sampleRate);

  // give no error back
  return(0);
}
